components/verify/Verifier.tsx

```diff
@@ -66,13 +66,29 @@ const Verifier: React.FC<VerifierProps> = ({ wrappedDocument }) => {
 
       setFragmentVerificationStatus({ DOCUMENT_STATUS, DOCUMENT_INTEGRITY, ISSUER_IDENTITY });
 
+      const docStatusFragments = fragments.filter(
+        (fragment) => fragment.type === "DOCUMENT_STATUS" && fragment.status === "INVALID"
+      ) as InvalidVerificationFragment<{}>[];
+      const isRevoked = docStatusFragments.some((fragment) => fragment.reason.codeString === "DOCUMENT_REVOKED");
```

Comment:
As discussed, I find Method 2 looks better too. Assuming that `revokedOnAny` is already evaluated correctly in `oa-verify`, I'd also think your updated Method 2 looks cleaner than the one you sent me.

My only suggestion is whether this whole function could be added into `utils` in the `oa-verify` package as a function with something like `isRevoked(fragments)` instead. I thought it might be a useful utility if checking for revocation is a common chore!😀

Reply:
One downside of Method 2 is that the typings do not work at this moment.

This is because the `isInvalidFragment()` type guard ([line 142-146](https://github.com/Open-Attestation/oa-verify/blob/610adb4a0fefad2f84e235a6e87ca3a78d61f2df/src/common/utils.ts#L142-L146)) will only extend the type with: `{ status: "INVALID" }` ([line 134](https://github.com/Open-Attestation/oa-verify/blob/610adb4a0fefad2f84e235a6e87ca3a78d61f2df/src/common/utils.ts#L134)). It doesn't extend the type any further with typings for `reason` or `data` like [this](https://github.com/Open-Attestation/oa-verify/blob/610adb4a0fefad2f84e235a6e87ca3a78d61f2df/src/types/core.ts#L63-L67).

Not too sure what's the best way to implement such utils 🤔

Reply:
Is there a reason for the inconsistency in the types for invalid fragments where some has `reason` and `data` while others only `status`? How about making them consistent?

Reply:
Not too sure either, Kyle suggests we could deprecate and create a new type in `oa-verify` then update the downstream repos to avoid breaking anything. But for now i'm going to merge this PR in first

Reply:
Sounds good.👍